src/routeGeneration/routeGenerator.ts

```diff
@@ -122,8 +122,13 @@ export class RouteGenerator {
   }
 
   private getRelativeImportPath(fileLocation: string) {
-    fileLocation = fileLocation.replace('.ts', '');
-    return `./${path.relative(this.options.routesDir, fileLocation).replace(/\\/g, '/')}`;
+    const outputDirectory =
+      path.extname(this.options.routesDir) === '' ? // routes dir is a directory
```

Comment:
@HoldYourWaffle, I agree that this needs unit tests. If you make the `buildContent` function a public function, you could write a test that checks if the handlebars output string contains the correct authenticationModule. Like did it find the path?
```
{{#if authenticationModule}}
import { koaAuthentication } from '{{authenticationModule}}';
{{/if}}
```

Reply:
It should also be possible to write a test that does something similar to `generate-routes.ts`.. so instantiate the `RouteGenerator` with specific options (`routesDir` as file and `routesDir` as dir) and then check the resulting file. This way the function could stay private and we would additionally test the public interface.